File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.py
def solution(cacheSize, cities):
    answer = 0
    stack = []
    for i in cities:
        i = i.lower()
        if i in stack:
            answer += 1
            stack.remove(i)
            stack.insert(0,i)
        else:
            answer += 5
        
            if cacheSize > 0:
                stack.insert(0,i)
                if len(stack) > cacheSize:
                    stack.pop()

    return answer

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(int cacheSize, vector<string> cities) {
    int answer = 0;
    vector<string> stack;

    for (string i : cities) {
        transform(i.begin(), i.end(), i.begin(), ::tolower);

        auto it = find(stack.begin(), stack.end(), i);

        if (it != stack.end()) {
            answer += 1;
            stack.erase(it);
            stack.insert(stack.begin(), i);
        }
        else {
            answer += 5;

            if (cacheSize > 0) {
                stack.insert(stack.begin(), i);

                if (stack.size() > cacheSize) {
                    stack.pop_back();
                }
            }
        }
    }

    return answer;
}